common/src/main/java/com/wynntils/wc/custom/item/UnidentifiedItemStack.java

```diff
@@ -68,30 +69,49 @@ public UnidentifiedItemStack(ItemStack stack) {
         if (rarityMap == null) return;
 
         List<String> items = rarityMap.get(tier);
-        if (items == null) return;
+        if (items == null || items.isEmpty()) return;
+
+        tooltip.add(new TranslatableComponent("feature.wynntils.itemGuess.possibilities"));
+
+        Map<Integer, List<MutableComponent>> levelToItems = new TreeMap<>();
 
-        MutableComponent guesses = new TextComponent("");
         for (String item : items) {
+            ItemProfile profile = WebManager.getItemsMap().get(item);
+
+            int level = (profile != null) ? profile.getLevelRequirement() : -1;
+
             MutableComponent itemDesc = new TextComponent(item).withStyle(tier.getChatFormatting());
 
-            ItemProfile itemProfile = WebManager.getItemsMap().get(item);
-            if (ItemGuessFeature.showGuessesPrice && itemProfile != null) {
-                int level = itemProfile.getLevelRequirement();
-                int itemCost = tier.getItemIdentificationCost(level);
-                itemDesc.append(new TextComponent(" [")
-                        .append(new TextComponent(itemCost + " " + EmeraldSymbols.E_STRING)
+            levelToItems.computeIfAbsent(level, i -> new ArrayList<>()).add(itemDesc);
+        }
+
+        for (Map.Entry<Integer, List<MutableComponent>> entry : levelToItems.entrySet()) {
+            int level = entry.getKey();
+            List<MutableComponent> itemsForLevel = entry.getValue();
+
+            MutableComponent guesses = new TextComponent("    ");
+
+            guesses.append(
+                    new TranslatableComponent("feature.wynntils.itemGuess.levelLine", level != -1 ? level : "?"));
+
+            if (ItemGuessFeature.showGuessesPrice && level != -1) {
+                guesses.append(new TextComponent(" [")
+                        .append(new TextComponent(
+                                        (tier.getItemIdentificationCost(level) + " " + EmeraldSymbols.E_STRING))
                                 .withStyle(ChatFormatting.GREEN))
                         .append(new TextComponent("]"))
                         .withStyle(ChatFormatting.GRAY));
             }
 
-            if (guesses.getSiblings().size() > 0)
-                guesses.append(new TextComponent(", ").withStyle(ChatFormatting.GRAY));
+            guesses.append("§7: ");
 
-            guesses.append(itemDesc);
-        }
+            guesses.append(itemsForLevel.stream()
+                    .reduce((i, j) -> i.append(new TextComponent(", ").withStyle(ChatFormatting.GRAY))
+                            .append(j))
+                    .get());
```

Comment:
If `itemsForLevel` is empty, reduce produces an empty optional, meaning the .get will fail. Perhaps add an `orElse` or a `isPresent` check.

(@magicus I believe this is the first time Qodana has pointed out something useful, although yes, it can and is stupid a lot of times)

Reply:
I don't think `itemsForLevel` can be empty as any creation of an entry in the TreeMap also adds an element to the list.

Reply:
Then what happens if there is only 1 item?

Reply:
I think it still works, but I will add a check nonetheless.

Reply:
Thanks!